docs/contributing/999-rfcs/2023-10-01-workloads.md

```diff
@@ -0,0 +1,262 @@
+   container code can just `import` the `@winglang/sdk` type definitions.
+1. Call any SDK and user-defined Wing inflight API from TypeScript/JavaScript (P2). This means that
+   we will need to inject the inflight client of Wing classes into the container during build and
+   somehow emit typescript type definitions for development (another reason why the Wing compile
+   needs to perform the build).
+1. Call any Wing inflight API from other languages (P3): this is lower priority for now given the
+   technical complexity but it would be healthy to start thinking about the path forward here. This
+   could also be a multi-phased project where we start with something like a sidecar/child process
+   (with generated client bindings) and then compile our inflight clients to WebAssembly when the
+   ecosystem is mature enough (currently there is no AWS SDK for WebAssembly). 
+
+The bindings APIs will need to also allow the user to specify the qualification of the bind (e.g.
+the permission requirements) so that the relevant permissions can be created.
+
+### Implementation for `cloud.Service` (P1)
+
+> Reqtag: `c:cloud-service`
+
+The `cloud.Service` resource exposes an API to implement a long running workload through a simple
+inflight closure:
+
+```js
+new cloud.Service(inflight () => {
+  log("service started");
+  return () => {
+    log("service stopped");
+  };
+});
+```
+
+This resource currently only has a simulator implementation. It bundles the inflight closure into a
+Node.js program and runs it within a sandbox. We currently don't have a cloud implementation for
+this resource, which is obviously a major gap.
+
+Once we have support for containerized workloads, it should be possible to implement this resource
+by synthesizing a Node.js docker image with the inflight closure code and a small wrapper (which
+invokes the start/stop callbacks), and deploy it as a container. Technically we would be able to
+remove the simulator implementation at that moment, because the simulator implementation of the
+container should be sufficient.
+
+This will also allow us to extend the `cloud.Service` API and support the various container
+capabilities such as exposing ports and adding API routes.
+
+### Implementation for `ex.Redis`, `ex.DynamoDb` (P2)
+
+> Reqtag: `c:ex-redis`
+
+We will also be able to implement other simulated resources such as `ex.Redis` and `ex.DynamoDb`
+using this new API and remove the duplicated container-management code from them.
+
+## Design Notes
+
+This section includes a few notes related to the design and implementation of these requirements.
+Consider them recommendations and pointers for the next phase of this project.
+
+### SDK resources
+
+I would recommend implementing this as a single "monolithic" resource called `cloud.Workload` which
+will expose all of the features described in this spec. This will produce a rich model for
```

Comment:
Checking if I understand this correctly - would `cloud.Service` be implemented on top of `cloud.Workload`? i.e. `cloud.Service` might package inflight code into a NodeJS dockerfile / image, and then pass that dockerfile / image to `cloud.Workload`?